core/bin/genesis_generator/src/main.rs

```diff
@@ -0,0 +1,113 @@
+    build::{prost_reflect, prost_reflect::ReflectMessage},
+    ProtoRepr,
+};
+use zksync_protobuf_config::proto::genesis::Genesis;
+use zksync_types::ProtocolVersionId;
+
+const DEFAULT_GENESIS_FILE_PATH: &'static str = "./etc/env/file_based//genesis.yaml";
+
+#[derive(Debug, Parser)]
+#[command(author = "Matter Labs", version, about = "zkSync operator node", long_about = None)]
+struct Cli {
+    #[arg(long)]
+    config_path: Option<std::path::PathBuf>,
+    #[arg(long, default_value = "false")]
+    check: bool,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let opt = Cli::parse();
+
+    let postgres_config = match opt.config_path {
+        None => PostgresConfig::from_env()?,
+        Some(path) => {
+            let yaml =
+                std::fs::read_to_string(&path).with_context(|| path.display().to_string())?;
+            let config =
+                decode_yaml_repr::<zksync_protobuf_config::proto::general::GeneralConfig>(&yaml)
+                    .context("failed decoding general YAML config")?;
+            config
+                .postgres_config
+                .context("Postgres config must exist")?
+        }
+    };
+
+    let yaml = std::fs::read_to_string(&DEFAULT_GENESIS_FILE_PATH)
+        .with_context(|| DEFAULT_GENESIS_FILE_PATH.to_string())?;
+    let original_genesis = decode_yaml_repr::<Genesis>(&yaml)?;
+    let db_url = postgres_config.master_url()?;
+    let new_genesis = generate_new_config(db_url, original_genesis.clone()).await?;
+    if opt.check {
+        assert_eq!(&original_genesis, &new_genesis);
+        println!("Genesis config is up to date");
+        return Ok(());
+    }
+    let data = encode_yaml(&Genesis::build(&new_genesis))?;
+    fs::write(DEFAULT_GENESIS_FILE_PATH, data)?;
+    Ok(())
+}
+
+async fn generate_new_config(
+    db_url: &str,
+    genesis_config: GenesisConfig,
+) -> anyhow::Result<GenesisConfig> {
+    let pool = ConnectionPool::<Core>::singleton(db_url)
+        .build()
+        .await
+        .context("failed to build connection_pool")?;
+    let mut storage = pool.connection().await.context("connection()")?;
+    let mut transaction = storage.start_transaction().await?;
+
+    if !transaction.blocks_dal().is_genesis_needed().await? {
+        anyhow::bail!("Please cleanup database for regenerating genesis")
+    }
+
+    let base_system_contracts = BaseSystemContracts::load_from_disk().hashes();
+    let mut updated_genesis = GenesisConfig {
+        protocol_version: Some(ProtocolVersionId::latest() as u16),
+        genesis_root_hash: None,
+        rollup_last_leaf_index: None,
+        genesis_commitment: None,
+        bootloader_hash: Some(base_system_contracts.bootloader),
+        default_aa_hash: Some(base_system_contracts.default_aa),
+        ..genesis_config
+    };
+
+    let params = GenesisParams::load_genesis_params(updated_genesis.clone())?;
+    let batch_params = insert_genesis_batch(&mut transaction, &params).await?;
```

Comment:
IMO, this tool shouldn't insert genesis batch, it should only generate config. I find it more handy, imagine you only want to regenerate config and check it but not modify DB. Anyway, if you disagree, you must at least mention somewhere in the description that this tool inserts genesis batch into DB.

Reply:
Yes, that's why i've not commit the tx. 
I wanted to reuse as much code as possible

Reply:
Added a comment about it